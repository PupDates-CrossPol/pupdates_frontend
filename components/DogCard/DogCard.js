import * as React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity} from 'react-native';
import { connect } from 'react-redux'
import { setPackInfo, setPackPhotos } from '../../actions'

const DogCard = (props) => {
    const dogPackCards = props.pack.map( (dog, i) => {
        
        const dogImages = props.packPhotos.map( (dogImage, i) =>  {
            return <Image key={i} source={{uri: dogImage.image_url}} style={styles.menuCircle}/>
        });
        
        return (
        <View key={i}>
            <Text>Name: {dog.name}</Text>
            <View style={styles.imageContainer}>
            
            </View>
            {dogImages}
            <Text>Sex: {dog.sex}</Text>
            <Text>Breed: {dog.breed}</Text>
            <Text>Size: {dog.size}</Text>
            <Text>Age:  {dog.age}</Text>
            <Text>Fixed: {dog.fixed ? 'true' : 'false'}</Text>
            <Text>Vaccinated: {dog.vaccinated ? 'true' : 'false'}</Text>
            <Text>Good With Kids: {dog.good_with_kids ? 'true' : 'false'}</Text>
        </View>
        )
    })
    
    return (
        <View><Text>THIS IS WHERE THE DOGS GO!!</Text>
       {dogPackCards}
        </View>
        )

}

const styles = StyleSheet.create({
    menuCircle: {
      aspectRatio: 1/1,
      height: 300,
      width: 300,
      borderRadius: 100,
      borderColor: 'black',
      borderWidth: 2,
    },
   
  });

export const mapStateToProps = state => ({
    user: state.user,
    pack: state.pack,
    packPhotos: state.packPhotos
  })
  
  export const mapDispatchToProps = dispatch => ({
    setPackInfo: (dogPack) => dispatch(setPackInfo(dogPack)),
    setPackPhotos: (dopPackPictures) => dispatch(setPackPhotos(dopPackPictures))
  })
  
  export default connect (mapStateToProps, mapDispatchToProps)(DogCard)