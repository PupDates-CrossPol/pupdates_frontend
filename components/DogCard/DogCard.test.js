import React from 'react'
import renderer from 'react-test-renderer'
import { DogCard } from './DogCard'

describe('<DogCard />', () => {
  it('has one child', () => {
    const mockPack = [
      {
        id: 1,
        user_id: 1,
        name: 'fluffy',
        breed: 'lab',
        size: 'medium',
        age: 7,
        good_with_kids: true,
        fixed: true,
        vaccinated: true
      }
    ]
  
    const mockPackPhotos = [
      {
        id: 1,
        dog_id: 1,
        image_url: 'some string'
      }
    ]
  
    const tree = renderer.create(<DogCard pack={mockPack} packPhotos={mockPackPhotos} />).toJSON()
    expect(tree.children.length).toBe(1)

  })
  it('renders correctly', () => {
    const mockPack = [
      {
        id: 1,
        user_id: 1,
        name: 'fluffy',
        breed: 'lab',
        size: 'medium',
        age: 7,
        good_with_kids: true,
        fixed: true,
        vaccinated: true
      }
    ]
  
    const mockPackPhotos = [
      {
        id: 1,
        dog_id: 1,
        image_url: 'some string'
      }
    ]
  
    const tree = renderer.create(<DogCard pack={mockPack} packPhotos={mockPackPhotos} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})