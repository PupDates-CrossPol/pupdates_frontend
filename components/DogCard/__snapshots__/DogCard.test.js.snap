// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DogCard /> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "borderBottomColor": "rgba(0,0,0,0.2)",
        "borderBottomWidth": 1,
        "flex": 1,
        "marginBottom": "5%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
          "marginBottom": "2%",
          "marginLeft": "7%",
        }
      }
    >
      fluffy
    </Text>
    <View
      style={
        Object {
          "borderColor": "rgba(0,0,0,0.2)",
          "borderRadius": 40,
          "borderWidth": 2,
          "flex": 1,
          "flexDirection": "row",
          "height": "50%",
          "marginBottom": "3%",
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": "8%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "some string",
              }
            }
            style={
              Object {
                "aspectRatio": 1,
                "borderColor": "rgba(0,0,0,0.2)",
                "borderRadius": 100,
                "borderWidth": 2,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.8,
                "shadowRadius": 2,
                "width": "90%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": "8%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "aspectRatio": 1,
                "borderColor": "rgba(0,0,0,0.2)",
                "borderRadius": 100,
                "borderWidth": 2,
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": "8%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "aspectRatio": 1,
                "borderColor": "rgba(0,0,0,0.2)",
                "borderRadius": 100,
                "borderWidth": 2,
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": "8%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "aspectRatio": 1,
                "borderColor": "rgba(0,0,0,0.2)",
                "borderRadius": 100,
                "borderWidth": 2,
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": "8%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "aspectRatio": 1,
                "borderColor": "rgba(0,0,0,0.2)",
                "borderRadius": 100,
                "borderWidth": 2,
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "margin": "8%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "aspectRatio": 1,
                "borderColor": "rgba(0,0,0,0.2)",
                "borderRadius": 100,
                "borderWidth": 2,
                "justifyContent": "center",
                "opacity": 1,
                "width": "90%",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View />
    <Text
      style={
        Object {
          "alignSelf": "stretch",
          "color": "grey",
          "fontSize": 20,
          "marginLeft": "7%",
          "textAlign": "left",
        }
      }
    >
      Sex: 
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "stretch",
          "color": "grey",
          "fontSize": 20,
          "marginLeft": "7%",
          "textAlign": "left",
        }
      }
    >
      Breed: 
      lab
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "stretch",
          "color": "grey",
          "fontSize": 20,
          "marginLeft": "7%",
          "textAlign": "left",
        }
      }
    >
      Size: 
      medium
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "stretch",
          "color": "grey",
          "fontSize": 20,
          "marginLeft": "7%",
          "textAlign": "left",
        }
      }
    >
      Age:  
      7
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "stretch",
          "color": "grey",
          "fontSize": 20,
          "marginLeft": "7%",
          "textAlign": "left",
        }
      }
    >
      Fixed: 
      true
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "stretch",
          "color": "grey",
          "fontSize": 20,
          "marginLeft": "7%",
          "textAlign": "left",
        }
      }
    >
      Vaccinated: 
      true
    </Text>
    <Text
      style={
        Object {
          "alignSelf": "stretch",
          "color": "grey",
          "fontSize": 20,
          "marginBottom": "5%",
          "marginLeft": "7%",
          "textAlign": "left",
        }
      }
    >
      Good With Kids: 
      true
    </Text>
  </View>
</View>
`;
